test/xcsync.tests/Projects/DotnetTest.cs

```diff
@@ -35,7 +35,7 @@ public async Task GetTypes ()
 		try {
 			var project = await clrProject.OpenProject ().ConfigureAwait (false);
 			var types = typeService.QueryTypes ().ToList ();
-			Assert.Equal (expectedTypes, [.. types.Where (x => x?.ClrType is not null).Select (x => x?.ClrType).OrderBy (x => x)]);
+			Assert.Equal (expectedTypes, [.. types.Where (x => x is not null && x.ClrType is not null & x.IsInSource).Select (x => x?.ClrType).OrderBy (x => x)]);
```

Comment:
Why unravel the ?. here?

```suggestion
			Assert.Equal (expectedTypes, [.. types.Where (x => x is not null && x.ClrType is not null && x.IsInSource).Select (x => x?.ClrType).OrderBy (x => x)]);
```

Reply:
Because otherwise the analyzer still thought x could be null for x.IsInSource